nifi-nar-bundles/nifi-box-bundle/nifi-box-processors/src/main/java/org/apache/nifi/processors/box/PutBoxFile.java

```diff
@@ -0,0 +1,407 @@
+            SUBFOLDER_NAME,
+            CREATE_SUBFOLDER,
+            FILE_NAME,
+            CONFLICT_RESOLUTION,
+            CHUNKED_UPLOAD_THRESHOLD
+    ));
+
+    public static final Relationship REL_SUCCESS =
+            new Relationship.Builder()
+                    .name("success")
+                    .description("Files that have been successfully written to Box are transferred to this relationship.")
+                    .build();
+
+    public static final Relationship REL_FAILURE =
+            new Relationship.Builder()
+                    .name("failure")
+                    .description("Files that could not be written to Box for some reason are transferred to this relationship.")
+                    .build();
+
+    public static final Set<Relationship> RELATIONSHIPS = Collections.unmodifiableSet(new HashSet<>(asList(
+            REL_SUCCESS,
+            REL_FAILURE
+    )));
+
+    private static final int CONFLICT_RESPONSE_CODE = 409;
+    private static final int NOT_FOUND_RESPONSE_CODE = 404;
+
+    private volatile BoxAPIConnection boxAPIConnection;
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @Override
+    public List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final String filename = context.getProperty(FILE_NAME).evaluateAttributeExpressions(flowFile).getValue();
+        final long chunkUploadThreshold = context.getProperty(CHUNKED_UPLOAD_THRESHOLD)
+                .asDataSize(DataUnit.B)
+                .longValue();
+        final ConflictResolutionStrategy conflictResolution = ConflictResolutionStrategy.forValue(context.getProperty(CONFLICT_RESOLUTION).getValue());
+
+        final long startNanos = System.nanoTime();
+        String fullPath = null;
+
+        try {
+            final long size = flowFile.getSize();
+            final BoxFolder parentFolder = getOrCreateDirectParentFolder(context, flowFile);
+            fullPath = BoxFileUtils.getFolderPath(parentFolder.getInfo());
+            BoxFile.Info uploadedFileInfo = null;
+
+            try (InputStream rawIn = session.read(flowFile)){
+
+                if (REPLACE.equals(conflictResolution)) {
+                    uploadedFileInfo = replaceExistingBoxFile(parentFolder, filename, rawIn, size, chunkUploadThreshold);
+                }
+
+                if (uploadedFileInfo == null) {
+                    if (size > chunkUploadThreshold) {
+                        uploadedFileInfo = parentFolder.uploadLargeFile(rawIn, filename, size);
+                    } else {
+                        uploadedFileInfo = parentFolder.uploadFile(rawIn, filename);
+                    }
```

Comment:
It might make sense extracting it into `createBoxFile()` (similar to the replace case).